File: assiut-sheets-1-python/sheet9/twins.py
n = int(input())
cs = [int(num) for num in input().split()]

cs.sort(reverse=True)
coin_sum = sum(cs)
sum1 = 0
sum2 = 0
count = 0

for c in cs:
    if sum1 <= coin_sum // 2:
        sum1 += c
        count += 1
    else:
        sum2 += c

print(count)

File: 900/twins.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    int n, sum1 = 0, sum2 = 0, count = 0;
    cin >> n;
    int a[n];
    for (int i = 0; i < n; i++) {
        cin >> a[i];
        sum1 += a[i];
    }
    sort(a, a+n);
    for (int i = n-1; i > -1; i--) {
        sum2 += a[i];
        count++;
        if (sum2 > sum1 / 2) break;
    }
    cout << count;
    return 0;
}